### ISSUE TITLE
Only using `unset apps` and `unset config` together will do the expected `unset apps`' job

### ISSUE THREAD

**Issue by MonicaisHer** [canonical/edgex-snap-hooks#43]
set app, unset apps, does not work:
```
$ sudo snap set edgex-device-mqtt apps.device-mqtt.config.service.port=1111
$ sudo snap restart edgex-device-mqtt.device-mqtt

$ sudo snap unset edgex-device-mqtt apps.device-mqtt.config.service.port
$ sudo snap restart edgex-device-mqtt.device-mqtt

$ snap get edgex-device-mqtt -d
{
    "apps": {
        "device-mqtt": {
            "config": {
                "service": {
                    "port": 1111
                }
            }
        }
    }
}
$ journalctl -f | grep device-mqtt |grep "Web server starting"
msg="Web server starting (localhost:1111)"
```

set app, unset apps and config, work as expected:
```
$ sudo snap set edgex-device-mqtt apps.device-mqtt.config.service.port=1111
$ sudo snap restart edgex-device-mqtt.device-mqtt

$ sudo snap unset edgex-device-mqtt apps.device-mqtt.config.service.port
$ sudo snap unset edgex-device-mqtt config.service.port
$ sudo snap restart edgex-device-mqtt.device-mqtt

$ snap get edgex-device-mqtt -d
{}
$ journalctl -f | grep device-mqtt |grep "Web server starting"
msg="Web server starting (localhost:59982)" <----------------------------------------default service port
```
See more details in this [PR](https://github.com/canonical/edgex-snap-testing/pull/48)

**Comment by farshidtz** [comment]
This is because the unset makes the config object empty and as a result the env file never gets emptied:
https://github.com/canonical/edgex-snap-hooks/blob/7fdbcd41ba0b01d036607afdbb5d17f073cc5acf/options/options.go#L59-L62

**Comment by farshidtz** [comment]
Fixed by #41
